synapse/rest/client/v1/room.py

```diff
@@ -309,7 +311,14 @@ def __init__(self, hs):
         self.room_member_handler = hs.get_room_member_handler()
         self.auth = hs.get_auth()
 
-    async def inherit_depth_from_prev_ids(self, prev_event_ids) -> int:
+    def _copy_requester_and_override_user_id(self, requester, new_user_id):
+        serialized_requester = requester.serialize()
+        serialized_requester["user_id"] = new_user_id
+        new_requester = Requester.deserialize(self.store, serialized_requester)
+
+        return new_requester
```

Comment:
Is there a better way to copy this efficiently?

Reply:
TBH I think in this case we probably explicitly want to set the fields, as I don't think we want to copy over e.g. `device_id` and the like? So I think we can just do `create_requester(user_id, app_server_id=requester.app_server_id)`?